sklearn/model_selection/_search_successive_halving.py

```diff
@@ -66,6 +68,25 @@ class BaseSuccessiveHalving(BaseSearchCV):
     Zohar Karnin, Tomer Koren, Oren Somekh
     """
 
+    _parameter_constraints: dict = {
+        **BaseSearchCV._parameter_constraints,
+        # overwrite `scoring` since multi-metrics are not supported
+        "scoring": [StrOptions(set(get_scorer_names())), callable, None],
+        "random_state": ["random_state"],
+        "max_resources": [
+            Interval(Integral, 0, None, closed="neither"),
+            StrOptions({"auto"}),
+        ],
+        "resource": [str],
+        "factor": [Interval(Real, 0, None, closed="neither")],
+        "min_resources": [
+            Interval(Integral, 0, None, closed="neither"),
+            StrOptions({"exhaust", "smallest"}),
+        ],
+        "aggressive_elimination": ["boolean"],
```

Comment:
It seems that some of the parameters there aren't sorted like the ones in the constructor's signature. Should we respect the order?